x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_fetch_alerts.tsx

```diff
@@ -196,19 +217,44 @@ const useFetchAlerts = ({
                   } else if (rawResponse.hits.total && typeof rawResponse.hits.total === 'object') {
                     totalAlerts = rawResponse.hits.total?.value ?? 0;
                   }
-                  dispatch({
-                    type: 'response',
-                    alerts: rawResponse.hits.hits.reduce<EcsFieldsResponse[]>((acc, hit) => {
-                      if (hit.fields) {
-                        acc.push({
-                          ...hit.fields,
-                          _id: hit._id,
-                          _index: hit._index,
-                        } as EcsFieldsResponse);
-                      }
+                  const alerts = rawResponse.hits.hits.reduce<EcsFieldsResponse[]>((acc, hit) => {
+                    if (hit.fields) {
+                      acc.push({
+                        ...hit.fields,
+                        _id: hit._id,
+                        _index: hit._index,
+                      } as EcsFieldsResponse);
+                    }
+                    return acc;
+                  }, []);
 
+                  const { oldAlertsData, ecsAlertsData } = alerts.reduce<{
+                    oldAlertsData: Array<Array<{ field: string; value: string[] }>>;
+                    ecsAlertsData: unknown[];
+                  }>(
+                    (acc, alert) => {
+                      const itemOldData = Object.entries(alert).reduce<
+                        Array<{ field: string; value: string[] }>
+                      >((oldData, [key, value]) => {
+                        oldData.push({ field: key, value: value as string[] });
+                        return oldData;
+                      }, []);
+                      const ecsData = Object.entries(alert).reduce((ecs, [key, value]) => {
```

Comment:
couldn't we create both objects at the same time avoiding running through it twice?

Reply:
I guess we should be able to. I can change it.. But I will wait @XavierM confirmation on this as he has added this piece of code.

Reply:
@jcger 

On second thoughts, I am not planning touch this before FF since it is working fine.  Do you think this can be handled with https://github.com/elastic/kibana/pull/143032

Reply:
Maybe a new tech debt issue? let's move on, we'll improve it if we see there are performance issues

Reply:
oh yes, we can do that, good idea?